pkg/volume:  use utils/ptr package instead of utils/pointer

liangyuanpeng opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

<!--
Add one of the following kinds:
/kind bug
/kind cleanup
/kind documentation
/kind feature

Optionally add one or more of the following kinds if applicable:
/kind api-change
/kind deprecation
/kind failing-test
/kind flake
/kind regression
-->
/kind cleanup

#### What this PR does / why we need it:

The new k8s.io/utils/ptr package provides generic wrapper functions, which can be used instead of type-specific pointer wrapper functions. This replaces the latter with the former, and migrates other uses of the deprecated pointer package to ptr in affacted files.

See kubernetes/utils#283 for details.

It will also reduce the friction with linter in the future

also releated:
- https://github.com/kubernetes/kubernetes/issues/132086

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

carlory commented:
/release-note-none
/ok-to-test
/lgtm

carlory commented on pkg/volume/csi/nodeinfomanager/nodeinfomanager_test.go:
```diff
@@ -41,7 +41,7 @@ import (
 	volumetest "k8s.io/kubernetes/pkg/volume/testing"
 	"k8s.io/kubernetes/pkg/volume/util"
-	utilpointer "k8s.io/utils/pointer"
+	ptr "k8s.io/utils/ptr"
```
please remove unnecessary alias.

liangyuanpeng replied:
done

carlory commented:
/lgtm
/assign @saad-ali

carlory reviewed (commented):
/assign @xing-yang

For approval

Octopusjust commented:
Could you replace the whole pointer wrapper functions in `pkg/volume` ?

liangyuanpeng committed:
pkg/volume: drop pointer wrapper functions.

The new k8s.io/utils/ptr package provides generic wrapper functions,
    which can be used instead of type-specific pointer wrapper functions.
    This replaces the latter with the former, and migrates other uses of
    the deprecated pointer package to ptr in affacted files.

    See kubernetes/utils#283 for details.

Signed-off-by: Lan Liang <gcslyp@gmail.com>

liangyuanpeng commented:
> Could you replace the whole pointer wrapper functions in pkg/volume ?

updated for some new code still uses old packages.

Octopusjust reviewed (commented):
/test pull-kubernetes-unit

liangyuanpeng commented:
/reopen

Rebased.

/cc @xing-yang  

Could you please review it? thanks.

liangyuanpeng commented:
/reopen
